src/Npgsql/Replication/PgOutput/Messages/TransactionChangingPgOutputReplicationMessage.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using NpgsqlTypes;
+
+namespace Npgsql.Replication.PgOutput.Messages
+{
+    /// <summary>
+    /// The common base class for all replication messages
+    /// that set the transation xid of a transaction
+    /// </summary>
+    public abstract class TransactionChangingPgOutputReplicationMessage : PgOutputReplicationMessage
```

Comment:
We don't even have "Replication" in the other subclasses so I removed that part too.
Shorter is definitely better and if that clutters someone's namespace or causes confusion it will happen with the other classes as well. You can't solve every problem by making class names longer. That's what namespaces are for.

Reply:
Agreed.